src/Native/System.Native/pal_console.cpp

```diff
@@ -35,9 +35,10 @@ extern "C" int32_t GetWindowSize(WinSize* windowSize)
 #endif
 }
 
-extern "C" int32_t IsATty(int fd)
+extern "C" int32_t IsATty(intptr_t fd)
 {
-    return isatty(fd);
+    assert(INT32_MIN <= fd && fd <= INT32_MAX);
```

Comment:
Also - the maximum FD isn't necessarily INT32_MAX (I believe). On OS X the max is OPEN_MAX in sys/syslimits.h. We should try to use these constants in our asserts if Ubuntu has a similar constant